userspace/libsinsp/sinsp.cpp

```diff
@@ -1864,10 +1864,12 @@ uint32_t sinsp::reserve_thread_memory(uint32_t size)
 
 void sinsp::get_capture_stats(scap_stats* stats) const
 {
-	if(scap_get_stats(m_h, stats) != SCAP_SUCCESS)
-	{
-		throw sinsp_exception(scap_getlasterr(m_h));
-	}
+	scap_get_stats(m_h, stats);
+}
+
+struct scap_stats_v2* sinsp::get_capture_stats_v2(uint32_t flags, uint32_t* nstats, int32_t* rc) const
+{
+	return scap_get_stats_v2(m_h, flags, nstats, rc);
```

Comment:
Same, we probably want to check `rc` somewhere. IMHO it should happen here, without having the callers care about `rc` at all.

Reply:
since the new approach is experimental I would avoid killing the whole process if something goes wrong, in the end, Falco can live also without stats :) Btw no strong opinion on that, it seemed a reasonable assumption

Reply:
I totally agree, let's not crash the process if we can't get the stats but IMO this is not the place. IMO this decision belongs in the caller (we can easily pokemon-handle the exceptions there) but exposing C-style error handling in a C++ lib doesn't feel right.

Reply:
See above comment something with the return error was anyways wrong in the old stats, we could defer this to a future cleanup?